### ISSUE TITLE
Can't fetch stderr of a job with container exit code 1

### ISSUE THREAD

**Issue by enricorotundo** [filecoin-project/bacalhau#504]
ETA: 2022Q4

Running an invalid command (resulting in container exit code 1) [should return stderr/stdout](https://github.com/filecoin-project/bacalhau/blob/5ec908fcdca2ac17344488c89d1fd2bf4d1c1d7b/pkg/executor/docker/executor.go#L296-L364). Instead, there's no results?

Example:

```zsh
❯ go run . --api-port=$BACALHAU_API_PORT_0 --api-host=localhost docker run ubuntu -- bash -c 'rm /'
cb635bf6-215a-4ca8-b6f2-85229a583095

❯ go run . --api-port=$BACALHAU_API_PORT_0 --api-host=localhost list --wide
 CREATION_TIME      ID                                    JOB                         STATE  RESULT 
 22-08-17-22:57:03  cb635bf6-215a-4ca8-b6f2-85229a583095  Docker ubuntu bash -c rm /  Error         

❯ go run . --api-port=$BACALHAU_API_PORT_0 --api-host=localhost get cb635bf6-215a-4ca8-b6f2-85229a583095 --output-dir ./enrico
[...]
[2207] INF bacalhau/get.go:35 > Fetching results of job 'cb635bf6-215a-4ca8-b6f2-85229a583095'...
[2207] DBG ipfs/downloader.go:42 > No results to download

❯ ls -l enrico
ls: enrico: No such file or directory
```

**Comment by js-ts** [comment]
Error: job has error state Error on node QmdZQ7ZbhnvWY1J12XYKGHApJ6aufKyLNSvf8jZBrBaAVL

I think that instead of returnining that a certain node, has error it should instead give the actual error

**Comment by enricorotundo** [comment]
Thread: https://filecoinproject.slack.com/archives/C02RLM3JHUY/p1660770923525699

**Comment by philwinder** [comment]
As a work around I've found that doing something like `-- bash -c "erroring-thing | true"` forces the execution to succeed and the stderr is returned.

But the fix is to return the stdout/err even upon failure.

**Comment by simonwo** [comment]
Fixed by #1545.
